### ISSUE TITLE
E0002 needs to be updated to new format

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/rust#35495]
From: src/test/compile-fail/E0002.rs

Errror E0002 needs a label and help moved to a simple help, updating it from:

```
error[E0002]: non-exhaustive patterns: type std::option::Option<i32> is non-empty
  --> src/test/compile-fail/E0002.rs:14:5
   |
14 |     match x { } //~ ERROR E0002
   |     ^^^^^^^^^^^
   |
help: Please ensure that all possible cases are being handled; possibly adding wildcards or more match arms.
  --> src/test/compile-fail/E0002.rs:14:5
   |
14 |     match x { } //~ ERROR E0002
   |     ^^^^^^^^^^^
```

To:

```
error[E0002]: non-exhaustive patterns: type std::option::Option<i32> is non-empty
  --> src/test/compile-fail/E0002.rs:14:5
   |
14 |     match x { } //~ ERROR E0002
   |     ^^^^^^^^^^^ not all cases are handled
   |
   = note: this may be fixed by possibly adding wildcards or more match arms.
```

**Comment by srdja** [comment]
I can work on this.

**Comment by arielb1** [comment]
This error is basically a hack and should be merged into E0004.

**Comment by bstrie** [comment]
I've opened #36724 to address @arielb1's comment. In the meantime, I'm leaving it up to @jonathandturner as to whether this issue should be addressed as normal or if we should close this outright. In the meantime, I'm operating under the assumption that this will remain open, and am offering it to anyone as a part of https://github.com/rust-lang/rust/issues/35233 .

**Comment by jfirebaugh** [comment]
#36909 has been merged and E0002 is now unused (modulo #37058), so this can be closed.

**Comment by jonathandturner** [comment]
Let's go ahead and close those.  We'll likely end up doing a sweep over the errors once the last errors are update and we move many of the compile-fail tests over to ui tests.
